File: Data Structures & Algorithms/two-integer-sum/submission-8.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        Arrays.sort(nums);
        int l=0,r=nums.length-1;
        
        while(l<r){
            int sum = nums[l] + nums[r];
            if(sum>target){
                r--;
            }else if(sum < target){
                l++;
            }else {
                return new int[] {l,r};
            }

        }
        
        return null;
    }
}

File: Data Structures & Algorithms/minimum-size-subarray-sum/submission-8.py
class Solution:
    def minSubArrayLen(self, target: int, nums: List[int]) -> int:
        ret = float('inf')         

        l,r = 0,0
        currSum = nums[0]

        while True:
            if currSum < target:
                r += 1
                if r >= len(nums):
                    return ret if ret < float('inf') else 0
                currSum += nums[r]
            else:
                ret = min(ret, r-l+1)
                currSum -= nums[l]
                l+=1
        
        return -1